src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -274,19 +279,28 @@ public void runAppCfgCommand(List<String> args) throws ProcessRunnerException {
   }
 
   /**
-   * Returns the version of the Cloud SDK installation. Unlike other methods in this class that call
-   * gcloud, this method always uses a synchronous ProcessRunner and will block until the gcloud
-   * process returns.
+   * Returns the version of the Cloud SDK installation.
    *
-   * @throws ProcessRunnerException when process runner encounters an error
+   * @throws CloudSdkOutOfDateException if the Cloud SDK is too out of date to determine its version
    */
-  public CloudSdkVersion getVersion() throws ProcessRunnerException {
-    // gcloud info --format="value(basic.version)"
-    List<String> command = new ImmutableList.Builder<String>().add("info")
-        .addAll(GcloudArgs.get("format", "value(basic.version)")).build();
+  public CloudSdkVersion getVersion() throws IOException {
+    Path versionFile = getSdkPath().resolve(VERSION_FILE);
 
-    String output = runSynchronousGcloudCommand(command);
-    return new CloudSdkVersion(output);
+    if (!Files.isRegularFile(versionFile)) {
+      throw new CloudSdkOutOfDateException("Cloud SDK version is out of date. Please update to "
```

Comment:
I'd change this warning to 'Cloud SDK versions below %MINIMUM_VERSION% are not supported by this library.'

'Cloud SDK version is 'out of date'' is too general in the sense that it could be out of date but supported.